lib/vehicle.rb

```diff
@@ -1,18 +1,20 @@
 require 'date'
 
 class Vehicle
-  attr_reader :vin,
-              :year,
-              :make,
-              :model,
-              :engine
+  attr_accessor :vin, :year, :make, :model, :engine, :registration_date, :plate_type
 
   def initialize(vehicle_details)
     @vin = vehicle_details[:vin]
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @plate_type = nil
+    @registration_date = nil
+  end
+
+  def registration_date
+    @registration_date
   end
 
```

Comment:
Since you have a method in your attr_accessor's for `registration_date`, there's no need for this method to return the attribute.